src/maker/config.rs

```diff
@@ -49,3 +52,186 @@ impl MakerConfig {
         }
     }
 }
+
+impl MakerConfig {
+    /// new a default configuration with given port and address
+    pub fn init(port: u16, onion_addrs: String) -> Self {
+        Self {
+            port,
+            onion_addrs,
+            ..MakerConfig::default()
+        }
+    }
+    pub fn new(file_path: Option<&PathBuf>) -> Self {
+        let default_config = Self::default();
+        let default_path = PathBuf::from("maker.toml");
+        let path = file_path.unwrap_or(&default_path);
+
+        // Check if the file exists
+        if !path.exists() {
+            eprintln!("Config file not found: {:?}", path);
+            return default_config;
+        }
+
+        // Attempt to parse the TOML file
+        let sections = match parse_toml(path) {
+            Ok(sections) => sections,
+            Err(err) => {
+                eprintln!("Error parsing config file: {:?}", err);
+                return default_config;
+            }
```

Comment:
What you would wanna do here is pass the error to the return of the `new` function with a `?`. To do this, the `new` function should return `Result<Self, std::io::Error>`, or `io::Result`, whatever feels good to you. 

Then, these match cases won't be required.